src/Symfony/Component/Validator/Constraints/IsinValidator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Laurent Masforné <l.masforne@gmail.com>
+ *
+ * @see https://en.wikipedia.org/wiki/International_Securities_Identification_Number
+ */
+class IsinValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Isin) {
+            throw new UnexpectedTypeException($constraint, Isin::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = strtoupper($value);
+        $canonicalize = str_replace(' ', '', $value);
```

Comment:
I am not convinced that we should by default normalize the values to be normalized. In the worst scenario this will result in a loss of data when whitespaces are stripped here, but preserved in the original data meaning that any length constraint on the database level will raise issues (in the best case there will be an exception, but it gets worse if the database layer silently trims the data).

Reply:
yes you're right, i remove this line, and it will lauch a validator violation ;)